tests/test_distributions.py

```diff
@@ -119,7 +122,7 @@ def test_check_distribution_compatibility() -> None:
     # test the same distribution
     for key in EXAMPLE_JSONS:
         distributions.check_distribution_compatibility(
-            EXAMPLE_DISTRIBUTIONS[key], EXAMPLE_DISTRIBUTIONS[key]
+            EXAMPLE_DISTRIBUTIONS[key], distributions.json_to_distribution(EXAMPLE_JSONS[key])
```

Comment:
We need to create new objects to compare NaNs.

>The built-in containers typically assume identical objects are equal to themselves.

https://docs.python.org/3/reference/expressions.html#value-comparisons

```python
>>> x = float("nan")
>>> y = float("nan")
>>> x == x
False
>>> x == y
False
>>> [x] == [x]
True
>>> [x] == [y]
False
```
Note that `copy.deepcopy` does not create new object because `float("nan")` is immutable.
```python
>>> [x] == [copy.deepcopy(x)]
True
```